code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -389,79 +389,85 @@
 	if(uses_stored && !stored_gun)
 		return PROCESS_KILL
 
+	//Turrets can shoot up, but not down.
+	var/list/valid_turfs = list()
+	var/turf/temp = get_turf(src)
+	while(temp && (isspaceturf(temp) || temp == get_turf(src)))
+		valid_turfs["[temp.z]"] = temp
+		temp = temp.above()
+
 	var/list/targets = list()
-	for(var/mob/A as() in hearers(scan_range, base))
-		if(A.invisibility > SEE_INVISIBLE_LIVING)
-			continue
-
-		if(check_anomalies)//if it's set to check for simple animals
-			if(isanimal(A))
-				var/mob/living/simple_animal/SA = A
-				if(SA.stat || in_faction(SA)) //don't target if dead or in faction
-					continue
-				targets += SA
+	for(var/turf_z as() in valid_turfs)
+		var/turf/T = valid_turfs[turf_z]
+		for(var/mob/A as() in hearers(scan_range, T))
+			if(A.invisibility > SEE_INVISIBLE_LIVING)
 				continue
 
-		if(issilicon(A))
-			var/mob/living/silicon/sillycone = A
+			if(check_anomalies)//if it's set to check for simple animals
+				if(isanimal(A))
+					var/mob/living/simple_animal/SA = A
+					if(SA.stat || in_faction(SA)) //don't target if dead or in faction
+						continue
+					targets += SA
+					continue
 
-			if(ispAI(A))
-				continue
+			if(issilicon(A))
+				var/mob/living/silicon/sillycone = A
 
-			if(target_cyborgs && sillycone.stat != DEAD && iscyborg(sillycone))
-				targets += sillycone
-				continue
+				if(ispAI(A))
+					continue
 
-			if(sillycone.stat || in_faction(sillycone))
-				continue
+				if(target_cyborgs && sillycone.stat != DEAD && iscyborg(sillycone))
+					targets += sillycone
+					continue
 
-			if(iscyborg(sillycone))
-				var/mob/living/silicon/robot/sillyconerobot = A
-				if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == TRUE)
+				if(sillycone.stat || in_faction(sillycone))
 					continue
 
-			targets += sillycone
-			continue
+				if(iscyborg(sillycone))
+					var/mob/living/silicon/robot/sillyconerobot = A
+					if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == TRUE)
```

Comment:
```suggestion
					if((ROLE_SYNDICATE in faction) && sillyconerobot.emagged)
```
idk if that would runtime or not

Reply:
Assuming faction is a lazylist based on the lazy define usage, it would runtime. LAZYFIND would probably be better. This isn't my code though.

Reply:
It shouldn't runtime, check discord.